Update Guava to 32.0.1-jre (Modern Java only)

### Description

Need to update Guava dependency from `31.1-jre` to `32.0.1-jre` due to https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2023-2976

### A problem with upgrade for the Java 8 branch

It was observed that Guava `32.0.0-jre` and  `32.0.1-jre` exhibit a new and erroneous behaviour when executing a Java program with a custom system class loader that instantiates `com.google.common.cache.Cache` in its constructor. 

A but report has been submitted with Google Guava https://github.com/google/guava/issues/6565

Here is a stack trace:
```
Error occurred during initialization of VM
java.lang.BootstrapMethodError: java.lang.ExceptionInInitializerError
	at com.google.common.cache.CacheBuilder.<clinit>(CacheBuilder.java:240)
	at ua.com.fielden.platform.classloader.TgSystemClassLoader.<init>(TgSystemClassLoader.java:33)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.SystemClassLoaderAction.run(ClassLoader.java:2220)
	at java.lang.SystemClassLoaderAction.run(ClassLoader.java:2204)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.lang.ClassLoader.initSystemClassLoader(ClassLoader.java:1450)
	at java.lang.ClassLoader.getSystemClassLoader(ClassLoader.java:1430)
Caused by: java.lang.ExceptionInInitializerError
	at java.lang.invoke.BoundMethodHandle.<clinit>(BoundMethodHandle.java:830)
	at java.lang.invoke.LambdaForm.createIdentityForms(LambdaForm.java:1778)
	at java.lang.invoke.LambdaForm.<clinit>(LambdaForm.java:1833)
	at java.lang.invoke.DirectMethodHandle.makePreparedLambdaForm(DirectMethodHandle.java:231)
	at java.lang.invoke.DirectMethodHandle.preparedLambdaForm(DirectMethodHandle.java:194)
	at java.lang.invoke.DirectMethodHandle.preparedLambdaForm(DirectMethodHandle.java:183)
	at java.lang.invoke.DirectMethodHandle.make(DirectMethodHandle.java:89)
	at java.lang.invoke.MethodHandles$Lookup.getDirectMethodCommon(MethodHandles.java:1660)
	at java.lang.invoke.MethodHandles$Lookup.getDirectMethodNoSecurityManager(MethodHandles.java:1617)
	at java.lang.invoke.MethodHandles$Lookup.getDirectMethodForConstant(MethodHandles.java:1802)
	at java.lang.invoke.MethodHandles$Lookup.linkMethodHandleConstant(MethodHandles.java:1751)
	at java.lang.invoke.MethodHandleNatives.linkMethodHandleConstant(MethodHandleNatives.java:477)
	at com.google.common.cache.CacheBuilder.<clinit>(CacheBuilder.java:240)
	at ua.com.fielden.platform.classloader.TgSystemClassLoader.<init>(TgSystemClassLoader.java:33)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.SystemClassLoaderAction.run(ClassLoader.java:2220)
	at java.lang.SystemClassLoaderAction.run(ClassLoader.java:2204)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.lang.ClassLoader.initSystemClassLoader(ClassLoader.java:1450)
	at java.lang.ClassLoader.getSystemClassLoader(ClassLoader.java:1430)
Caused by: java.lang.IllegalStateException: recursive invocation
	at java.lang.ClassLoader.initSystemClassLoader(ClassLoader.java:1444)
	at java.lang.ClassLoader.getSystemClassLoader(ClassLoader.java:1430)
	at sun.invoke.util.BytecodeDescriptor.parseMethod(BytecodeDescriptor.java:47)
	at sun.invoke.util.BytecodeDescriptor.parseMethod(BytecodeDescriptor.java:41)
	at java.lang.invoke.MethodType.fromMethodDescriptorString(MethodType.java:1068)
	at java.lang.invoke.BoundMethodHandle$Factory.makeCbmhCtor(BoundMethodHandle.java:818)
	at java.lang.invoke.BoundMethodHandle$Factory.makeCtors(BoundMethodHandle.java:763)
	at java.lang.invoke.BoundMethodHandle$SpeciesData.initForBootstrap(BoundMethodHandle.java:361)
	at java.lang.invoke.BoundMethodHandle$SpeciesData.<clinit>(BoundMethodHandle.java:426)
	at java.lang.invoke.BoundMethodHandle.<clinit>(BoundMethodHandle.java:830)
	at java.lang.invoke.LambdaForm.createIdentityForms(LambdaForm.java:1778)
	at java.lang.invoke.LambdaForm.<clinit>(LambdaForm.java:1833)
	at java.lang.invoke.DirectMethodHandle.makePreparedLambdaForm(DirectMethodHandle.java:231)
	at java.lang.invoke.DirectMethodHandle.preparedLambdaForm(DirectMethodHandle.java:194)
	at java.lang.invoke.DirectMethodHandle.preparedLambdaForm(DirectMethodHandle.java:183)
	at java.lang.invoke.DirectMethodHandle.make(DirectMethodHandle.java:89)
	at java.lang.invoke.MethodHandles$Lookup.getDirectMethodCommon(MethodHandles.java:1660)
	at java.lang.invoke.MethodHandles$Lookup.getDirectMethodNoSecurityManager(MethodHandles.java:1617)
	at java.lang.invoke.MethodHandles$Lookup.getDirectMethodForConstant(MethodHandles.java:1802)
	at java.lang.invoke.MethodHandles$Lookup.linkMethodHandleConstant(MethodHandles.java:1751)
	at java.lang.invoke.MethodHandleNatives.linkMethodHandleConstant(MethodHandleNatives.java:477)
	at com.google.common.cache.CacheBuilder.<clinit>(CacheBuilder.java:240)
	at ua.com.fielden.platform.classloader.TgSystemClassLoader.<init>(TgSystemClassLoader.java:33)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.SystemClassLoaderAction.run(ClassLoader.java:2220)
	at java.lang.SystemClassLoaderAction.run(ClassLoader.java:2204)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.lang.ClassLoader.initSystemClassLoader(ClassLoader.java:1450)
	at java.lang.ClassLoader.getSystemClassLoader(ClassLoader.java:1430)
```

The Modern Java branch of TG stopped using a custom system class loader, which should make it possible to upgrade to `32.0.1-jre`.

### Expected outcome
More stable and secure version of Guava.

--- a/platform-pojo-bl/pom.xml
+++ b/platform-pojo-bl/pom.xml
@@ -18,6 +18,7 @@
     <properties>
         <jackson.version>2.15.0</jackson.version>
         <log4j.version>2.17.2</log4j.version>
+        <guava.version>32.0.1-jre</guava.version>
     </properties>
 
     <dependencies>
@@ -31,7 +32,7 @@
         <dependency>
             <groupId>com.google.guava</groupId>
             <artifactId>guava</artifactId>
-            <version>31.1-jre</version>
+            <version>${guava.version}</version>
         </dependency>
         <dependency>
             <groupId>com.google.inject</groupId>

